lib/vehicle.rb

```diff
@@ -6,20 +6,35 @@ class Vehicle
               :make,
               :model,
               :engine
-
+  
+  attr_accessor   :registration_date,
+                  :plate_type
+ 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
```

Comment:
You never reference this anywhere but here. You built a method called `plate_type` below and that is the more dynamic way to go about it.